packages/dapp/src/components/Navbar/Menu/MainMenu/MainMenu.tsx

```diff
@@ -1,93 +1,99 @@
-import { useTranslation } from 'react-i18next';
 import {
-  SubMenuKeys,
-  useMainMenuItems,
-  useMainSubMenuDevelopers,
-  useMainSubMenuLanguage,
-  useMainSubMenuShowcases,
-  useMainSubMenuTheme,
-} from '../../../../const';
-import { useMenu } from '../../../../providers/MenuProvider';
-import { SupportModal } from '../../../SupportModal';
+  MenuKeys,
+  useDevelopersContent,
+  useLanguagesContent,
+  useMainMenuContent,
+  useShowcasesContent,
+  useThemeContent,
+} from '@transferto/dapp/src/const';
+import { useMenuStore } from '@transferto/dapp/src/stores';
+import { useTranslation } from 'react-i18next';
+import { shallow } from 'zustand/shallow';
 import { MenuItemComponent, NavbarMenu, SubMenuComponent } from '../../index';
 interface MainMenuProps {
   handleClose: (event: MouseEvent | TouchEvent) => void;
 }
 
 export const MainMenu = ({ handleClose }: MainMenuProps) => {
-  const menu = useMenu();
   const i18Path = 'navbar.';
   const { t: translate } = useTranslation();
-  const mainMenuItems = useMainMenuItems();
-  const mainSubMenuTheme = useMainSubMenuTheme();
-  const mainSubMenuDevelopers = useMainSubMenuDevelopers();
-  const mainSubMenuLanguage = useMainSubMenuLanguage();
-  const mainSubMenuShowcases = useMainSubMenuShowcases();
+  const mainMenuItems = useMainMenuContent();
+  const mainSubMenuTheme = useThemeContent();
+  const mainSubMenuDevelopers = useDevelopersContent();
+  const mainSubMenuLanguage = useLanguagesContent();
+  const mainSubMenuShowcases = useShowcasesContent();
+  const [
+    openMainNavbarMenu,
+    onOpenNavbarMainMenu,
+    openNavbarSubMenu,
+    onOpenNavbarSubMenu,
+  ] = useMenuStore(
+    (state) => [
+      state.openMainNavbarMenu,
+      state.onOpenNavbarMainMenu,
+      state.openNavbarSubMenu,
+      state.onOpenNavbarSubMenu,
+    ],
+    shallow,
+  );
 
-  return (
+  return openMainNavbarMenu ? (
     <>
       <NavbarMenu
         handleClose={handleClose}
-        open={menu.openMainNavbarMenu}
-        setOpen={menu.onOpenNavbarMainMenu}
-        isOpenSubMenu={menu.openNavbarSubMenu !== SubMenuKeys.none}
+        open={true}
+        transformOrigin={'top right'}
```

Comment:
Now, those menus open from a different origin:

Reacting to your comment @chybisov https://github.com/lifinance/jumper.exchange/pull/502#issuecomment-1512564376